### PR TITLE
fix: service ordering must be alphabetical

### LINKED ISSUE #780: Service ordering
As discussed in this issue from the forums: https://forum.dfinity.org/t/services-wont-deserialize-properly-if-functions-arent-in-alphabetical-order/20885/10 for some reason services must have their fields sorted into alphabetical order.

The current sorting for service fields is not alphabetical. Errors like the following thus occur: `method name query1 duplicate or not sorted`.

I am working on a PR to fix this.

### PR DESCRIPTION
**Pull request by lastmjs**
Closes #780

# Description

As discussed in [this forum post](https://forum.dfinity.org/t/services-wont-deserialize-properly-if-functions-arent-in-alphabetical-order/20885/10) service field ordering must be alphabetical or errors will be encountered during deserialization.

Fixes # 780

# How Has This Been Tested?

We ran into these problems while building Azle 0.18.0, and our automated tests for services were breaking before introducing this fix. We rely heavily on the JS agent now in Azle 0.18.0 and hundreds of our integration tests pass using this new code. We seem to be the only ones that I know of heavily testing/using Candid services directly, as we've found both the Rust and JS agent bugs for service ordering.

These tests ran on our fork of just the @dfinity/candid package, and I've simply copied the changes into a recently cloned fork of the agent-js. 

If you want to see the tests that we've run I can provide links to Azle's example integration tests.

# Checklist:

- [x] My changes follow the guidelines in [CONTRIBUTING.md](https://github.com/dfinity/agent-js/blob/main/CONTRIBUTING.md).
- [x] The title of this PR complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/).
- [ ] I have edited the CHANGELOG accordingly. (I don't understand how to do this, the changelog seems generated)
- [ ] I have made corresponding changes to the documentation. (Do you think there needs to be any documentation changes?

### CODE REVIEW

**Review by chenyan-dfinity — APPROVED** [review]
LGTM

File: packages/candid/src/idl.ts [hunk 1]
```diff
@@ -1481,7 +1481,15 @@ export class ServiceClass extends ConstructType<PrincipalId> {
   public readonly _fields: Array<[string, FuncClass]>;
   constructor(fields: Record<string, FuncClass>) {
     super();
-    this._fields = Object.entries(fields).sort((a, b) => idlLabelToId(a[0]) - idlLabelToId(b[0]));
+    this._fields = Object.entries(fields).sort((a, b) => {
+      if (a[0] < b[0]) {
```
Comment (peterpeterparker) [inline packages/candid/src/idl.ts:1485]: Randomly noticing this PR. Couldn't this be summarized by `a[0] - a[1]`?

**Comment by lastmjs** [comment]
Sorry I accidentally hit the re-request review button @chenyan-dfinity

**Comment by lastmjs** [comment]
On that note, is there anything else that needs to be done to get these tests running?

**Review by krpeacock — APPROVED** [review]

**Comment by krpeacock** [comment]
Someone from SDK just had to push a couple buttons. As long as CI passes it should automerge

**Comment by lastmjs** [comment]
Mitm is failing with some kind of connection refused error, not sure if it's related to the PR

**Comment by krpeacock** [comment]
@lastmjs can you rebase or merge main into your branch? It should fix the tests

**Pull request merged**